enumerable.rb

```diff
@@ -0,0 +1,122 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    each do |i|
+      yield i
+    end
+    self
+  end
+
+  def my_each_with_index
+    (0...to_a.length).each do |i|
+      yield(to_a[i], i)
+    end
+    self
+  end
+
+  def my_select
+    arr = []
+    to_a.my_each do |num|
+      yield num ? arr.push(num) : ''
+    end
+    arr
+  end
+
+  def my_all?
+    to_a.my_each do |num|
+      return false unless yield num
+    end
+    true
+  end
+
+  def my_any?
+    to_a.my_each do |num|
+      return true if yield num
+    end
+    false
+  end
+
+  def my_none?
+    if block_given?
+      to_a.my_any?(&Proc.new)
+    else
+      to_a.length.positive?
+    end
+  end
+
+  def my_count(arg = false)
+    if block_given?
+      return 'Error: arg and block not allowed at the same time' if arg
+
+      count = 0
+      to_a.my_each do |item|
+        count += 1 if yield item
+      end
+    else
+      return length unless arg
+
+      count = 0
+      to_a.my_each do |item|
+        count += 1 if item == arg
+      end
+      count
+    end
+    count
+  end
+
+  def my_map
+    arr = []
+    to_a.my_each do |item|
+      arr.push(yield item)
+    end
+    arr
+  end
+
+  def my_inject(arg = false)
+    if arg
+      acc = arg
+      (0...to_a.length).my_each { |i| acc = yield(acc, to_a[i]) }
+    else
+      acc = to_a[0]
+      (1...to_a.length).my_each { |i| acc = yield(acc, to_a[i]) }
+    end
+    acc
+  end
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] when a symbol is specified combines each element of the collection by applying the symbol as a named method